csrc/fips_status.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+#include <cstdio>
+#include <functional>
+#include <jni.h>
+
+#include "string_vector.h"
+
+static AmazonCorrettoCryptoProvider::ConcurrentStringVector fipsStatusErrors;
+
+// To have this symbol exported, one needs to modify the final-link.version and the CMakeLists.txt
+extern "C" void AWS_LC_fips_failure_callback(char const* message);
+
+#if defined(FIPS_SELF_TEST_SKIP_ABORT)
+void AWS_LC_fips_failure_callback(char const* message)
+{
```

Comment:
Should we add some defensive input validation here like a character limit to avoid printing out all the memory? Similarly, should we have some upper bound on the number of errors to store? I can see some errant host running into a situation where it pushes an infinite number of identical error messages until it OOMs over.

Reply:
128 sounds good. 1024 error messages seems excessive. Usually when you start piling up rare error cases, it's a lot of instances of the same thing.